tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,850 @@
+	}
+
+	if pt, ok := pluginTree.(*toml.Tree); ok {
+		return pt, nil
+	}
+	return nil, fmt.Errorf("plugin config is not a TOML tree")
+}
+
+// getRuntimesConfig gets the runtimes configuration from the plugin config
+func getRuntimesConfig(pluginConfig *toml.Tree, version int) (map[string]interface{}, error) {
+	containerdSection := pluginConfig.Get("containerd")
+	if containerdSection == nil {
+		return nil, fmt.Errorf("containerd section not found")
+	}
+
+	containerdTree, ok := containerdSection.(*toml.Tree)
+	if !ok {
+		return nil, fmt.Errorf("containerd section is not a TOML tree")
+	}
+
+	runtimes := containerdTree.Get("runtimes")
+	if runtimes == nil {
+		return nil, fmt.Errorf("runtimes section not found")
+	}
+
+	// Handle both map and *toml.Tree types
+	switch v := runtimes.(type) {
+	case map[string]interface{}:
+		return v, nil
+	case *toml.Tree:
+		return tomlTreeToMap(v), nil
+	default:
+		return nil, fmt.Errorf("runtimes is not a map or toml.Tree, got %T", runtimes)
+	}
+}
+
+// getCDIEnabled checks if CDI is enabled in the plugin configuration
+func getCDIEnabled(pluginConfig *toml.Tree, version int) (bool, error) {
+	cdiEnabled := pluginConfig.Get("enable_cdi")
+	if cdiEnabled == nil {
+		return false, nil // CDI not configured, default is false
+	}
+
+	if enabled, ok := cdiEnabled.(bool); ok {
+		return enabled, nil
+	}
+
+	return false, fmt.Errorf("enable_cdi is not a boolean")
+}
+
+// getDefaultRuntime gets the default runtime name from the containerd configuration
+func getDefaultRuntime(pluginConfig *toml.Tree) (string, error) {
+	containerdSection := pluginConfig.Get("containerd")
+	if containerdSection == nil {
+		return "", fmt.Errorf("containerd section not found")
+	}
+
+	containerdTree, ok := containerdSection.(*toml.Tree)
+	if !ok {
+		return "", fmt.Errorf("containerd section is not a TOML tree")
+	}
+
+	defaultRuntime := containerdTree.Get("default_runtime_name")
+	if defaultRuntime == nil {
+		return "", nil // No default runtime set
+	}
+
+	if runtime, ok := defaultRuntime.(string); ok {
+		return runtime, nil
+	}
+
+	return "", fmt.Errorf("default_runtime_name is not a string")
+}
+
+// validateRuntimeConfig validates a specific runtime configuration
+func validateRuntimeConfig(runtime interface{}, expectedType string, expectedOptions map[string]interface{}) error {
+	var runtimeMap map[string]interface{}
+
+	// Handle both map and *toml.Tree types
+	switch v := runtime.(type) {
+	case map[string]interface{}:
+		runtimeMap = v
+	case *toml.Tree:
+		runtimeMap = tomlTreeToMap(v)
+	default:
+		return fmt.Errorf("runtime is not a map or toml.Tree, got %T", runtime)
+	}
+
+	// Check runtime type
+	runtimeType, ok := runtimeMap["runtime_type"].(string)
+	if !ok {
+		return fmt.Errorf("runtime_type not found or not a string")
+	}
+
+	if expectedType != "" {
+		// For empty expectedType, we don't check the runtime type
+		// For nvidia runtime, we accept various runtime types
+		if expectedType == "nvidia" {
```

Comment:
We never seem to call this with `expectedType == "nvidia"` what is the purpose of this branch?